plugin.js

```diff
@@ -106,8 +126,9 @@ function plugin (fastify, options, next) {
   }
 
   fastify.decorateRequest('cookies', null)
-  fastify.decorateReply('cookie', setCookie)
+  fastify.decorateReply('setCookieSet', null)
```

Comment:
I would avoid a decorator because this `set` is an internal thing.
I would decorate the reply with a symbol (not sure we did it before) or simply `fastifyRes[symbold] = new Set()` upper in this file

Reply:
Agreed, I just couldn't figure out where to store the data without hurting performance

I also want to set the `Set-Cookie` header once using a hook instead of every time `cookie()` is called

Lastly, I believe this approach breaks a case where we set 2 cookies that share the same name but have different options like `path` or `domain`. For instance, we can set foo=foo for the path `/foo` and foo=bar for `/bar`. So that needs to be accounted for as well

Reply:
You can use s Symbol instead

Reply:
Thank you for the suggestions 🙏

I think we still need to check options like `path` and `domain` before updating a cookie though. For that, we either need to reparse the serialized cookie that shares the same `name`; do some string operations; or maybe we can keep them as objects to make these checks easier and serialize them during the preSerialization stage using another hook?